### PR TITLE
Make attribute `ingress_class_name` of resource `kubernetes_ingress_v1` computed

### PR DESCRIPTION
**Pull request by arybolovlev**
### Description

This PR fixes an issue when the provider removes the attribute `ingress_class_name` that is set by Kubernetes to the [default ingress class](https://kubernetes.io/docs/concepts/services-networking/ingress/#default-ingress-class) value if it does not explicitly define. The attribute `ingress_class_name` of resource `kubernetes_ingress_v1` is now marked as computed.

### Acceptance tests
- [X] Have you added an acceptance test for the functionality being added?
- [X] Have you run the acceptance tests on this branch?

Output from acceptance testing:

```console
$ make testacc TESTARGS="-count 1 -run ^TestAccKubernetesIngressV1_defaultIngressClass"
==> Checking that code complies with gofmt requirements...
go vet .
TF_ACC=1 go test "/Users/sacha/Documents/code/hashicorp/terraform-provider-kubernetes/kubernetes" -v -count 1 -run ^TestAccKubernetesIngressV1_defaultIngressClass -timeout 3h
=== RUN   TestAccKubernetesIngressV1_defaultIngressClass
--- PASS: TestAccKubernetesIngressV1_defaultIngressClass (2.12s)
PASS
ok  	github.com/hashicorp/terraform-provider-kubernetes/kubernetes	2.702s
```

### Release Note
Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-kubernetes/blob/main/CHANGELOG.md):
```release-note
`resource/kubernetes_ingress_v1`: make the attribute `spec.ingress_class_name` computed.
```

### References
- Fix: https://github.com/hashicorp/terraform-provider-kubernetes/issues/1946

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

File: kubernetes/resource_kubernetes_ingress_test.go [hunk 1]
```diff
@@ -185,21 +185,6 @@ func TestAccKubernetesIngress_WaitForLoadBalancerGoogleCloud(t *testing.T) {
 	})
 }
 
-func testAccCheckKubernetesIngressForceNew(old, new *api.Ingress, wantNew bool) resource.TestCheckFunc {
```
Comment (arybolovlev) [inline kubernetes/resource_kubernetes_ingress_test.go:188]: Not related to the original issue, just noticed that it is not in use, and since I touched this file, I cleaned it up a bit.

File: kubernetes/resource_kubernetes_ingress_test.go [hunk 2]
```diff
@@ -452,47 +437,3 @@ resource "kubernetes_ingress" "test" {
   wait_for_load_balancer = true
 }`, name, name, name, name, name, name, name)
 }
-
-func testAccKubernetesIngressConfig_stateUpgradev0(provider, name string) string {
```
Comment (arybolovlev) [inline kubernetes/resource_kubernetes_ingress_test.go:456]: Not related to the original issue, just noticed that it is not in use, and since I touched this file, I cleaned it up a bit.

File: kubernetes/resource_kubernetes_ingress_v1_test.go [hunk 3]
```diff
@@ -310,19 +310,33 @@ func TestAccKubernetesIngressV1_multipleRulesDifferentHosts(t *testing.T) {
 	})
 }
 
-func testAccCheckKubernetesIngressV1ForceNew(old, new *networking.Ingress, wantNew bool) resource.TestCheckFunc {
```
Comment (arybolovlev) [inline kubernetes/resource_kubernetes_ingress_v1_test.go:313]: Not related to the original issue, just noticed that it is not in use, and since I touched this file, I cleaned it up a bit.

File: website/docs/d/pod.html.markdown [hunk 4]
```diff
@@ -310,8 +310,8 @@ Please, look at the example below:
 
 ```hcl
 resources {
-    limits = {}
-    requests = {}
+  limits   = {}
```
Comment (arybolovlev) [inline website/docs/d/pod.html.markdown:313]: This and the rest of the `website/docs` changes are not related to the original issue, just fix for some issues found by linter.

**Review by alexsomesan — APPROVED** [review]
LGTM! Great that you spotted the dead test code and cleaned it up!
